google-cloud-firestore/lib/google/cloud/firestore/field_value.rb

```diff
@@ -107,6 +135,72 @@ def self.delete
         def self.server_time
           new :server_time
         end
+
+        ##
+        # Creates a sentinel value to indicate the union of the given values
+        # with an array.
+        #
+        # @param [Object] values The values to add from the array. Required.
+        #
+        # @return [FieldValue] The array union field value object.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   array_union = Google::Cloud::Firestore::FieldValue.array_union(
+        #     1, 2, 3
+        #   )
+        #   array_union.type #=> :array_union
+        #   array_union.values #=> [1, 2, 3]
+        #
+        #   nyc_ref.update({ name: "New York City",
+        #                    lucky_numbers: array_union })
+        #
+        def self.array_union *values
+          # We can flatten the values because arrays don't support sub-arrays
+          values.flatten!
+          raise ArgumentError, "values must be provided" if values.nil?
```

Comment:
Please note that you need to validate that there are no invalid field values in `values`. The first version that we shipped in Node silently dropped field value sentinels if there were contained inside field value sentinels. `FieldValue.arrayUnion(FieldValue.arrayUnion('foo'))` created an empty array. Instead, we should throw an error.

Reply:
Added this check, and tests to ensure that an error is raised.